adc/onboard_temperature/onboard_temperature.c

```diff
@@ -0,0 +1,59 @@
+/**
+ * Copyright (c) 2021 Raspberry Pi (Trading) Ltd.
+ *
+ * SPDX-License-Identifier: BSD-3-Clause
+ *
+ */
+
+#include <stdio.h>
+#include "pico/stdlib.h"
+#include "hardware/adc.h"
+
+/* Choose 'C' for Celsius or 'F' for Fahrenheit. */
+#define TEMPERATURE_UNITS 'C'
+
+/* References for this implementation:
+ * raspberry-pi-pico-c-sdk.pdf, Section '4.1.1. hardware_adc'
+ * pico-examples/adc/adc_console/adc_console.c */
+const float read_onboard_temperature(const char unit) {
+    
+    /* 12-bit conversion, assume max value == ADC_VREF == 3.3 V */
+    const float conversionFactor = 3.3f / (1 << 12);
+
+    float adc = adc_read() * conversionFactor;
+    float tempC = 27.0 - (adc - 0.706) / 0.001721;
+
+    if (unit == 'C') {
+        return tempC;
+    } else if (unit == 'F') {
+        return tempC * 9 / 5 + 32;
```

Comment:
Given that we're working with floats anyway, I guess this could be `return tempC * 1.8 + 32;` :thermometer:

Reply:
I tested with both, with same results. Ended up using `9/5` for clarity. :)